great_expectations/validator/v1_validator.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import TYPE_CHECKING, Optional
+
+from great_expectations.core.expectation_validation_result import (
+    ExpectationSuiteValidationResult,
+    ExpectationValidationResult,
+)
+from great_expectations.validator.validator import calc_validation_statistics
+
+if TYPE_CHECKING:
+    from great_expectations.core import ExpectationSuite
+    from great_expectations.core.batch_config import BatchConfig
+    from great_expectations.data_context import AbstractDataContext
+    from great_expectations.datasource.fluent.batch_request import BatchRequestOptions
+    from great_expectations.expectations.expectation import (
+        Expectation,
+        ExpectationConfiguration,
+    )
+
+
+class ResultFormat(str, Enum):
+    BOOLEAN_ONLY = "BOOLEAN_ONLY"
+    BASIC = "BASIC"
+    SUMMARY = "SUMMARY"
+    COMPLETE = "COMPLETE"
+
+
+class Validator:
+    """Validator.
+
+    Responsible for running expectations on a batch configuration.
+    """
+
+    def __init__(
+        self,
+        context: AbstractDataContext,
+        batch_config: BatchConfig,
+        result_format: ResultFormat = ResultFormat.SUMMARY,
+        batch_asset_options: Optional[BatchRequestOptions] = None,
+    ) -> None:
+        self._context = context
+        self._batch_config = batch_config
+        self._batch_asset_options = batch_asset_options
+        self.result_format = result_format
+
+    def validate_expectation(
+        self, expectation: Expectation
+    ) -> ExpectationValidationResult:
+        """Run a single expectation against the batch config"""
+        results = self._validate_expectation_configs([expectation.configuration])
+
+        assert len(results) == 1
+        return results[0]
+
+    def validate_expectation_suite(
+        self, expectation_suite: ExpectationSuite
+    ) -> ExpectationSuiteValidationResult:
+        """Run an  expectation suite against the batch config"""
+        results = self._validate_expectation_configs(expectation_suite.expectations)
+        statistics = calc_validation_statistics(results)
+
+        # TODO: This was copy/pasted from Validator, but many fields were removed
+        return ExpectationSuiteValidationResult(
+            results=results,
+            success=statistics.success,
+            statistics={
+                "evaluated_expectations": statistics.evaluated_expectations,
+                "successful_expectations": statistics.successful_expectations,
+                "unsuccessful_expectations": statistics.unsuccessful_expectations,
+                "success_percent": statistics.success_percent,
+            },
+        )
+
+    def _validate_expectation_configs(
+        self, expectation_configs: list[ExpectationConfiguration]
+    ) -> list[ExpectationValidationResult]:
+        """Run a list of expectation configurations against the batch config"""
+        batch_request = self._batch_config.data_asset.build_batch_request(
```

Comment:
We'll have to think about the API that a batch asset makes for splitting up the data into batches. We do have a lot of DataAssets and want adding more to be formulaic for us or contributors. Right now batching the data is determined by the splitter on the data asset. In the new world, the data asset is the data and the info about how you want to split it into batches lives in the batch config. Maybe this means `data_asset.build_batch_asset` will take in the splitters as well as the `options`? 

It feels like here, the validator should call a method on the batch config to get the `batch_request`, eg `self._batch_config.build_batch_request(...)`. Right now this would call the identical function on the asset vs reaching into the `self._batch_config.data_asset`. I feel this way because I think the splitters will live on `self._batch_config` and the validator code here will not need to explicitly know about splitters, it just wants a `batch_request`. 

The ask for now would be to move `build_batch_request` up to the `batch_config`. We can do that in a followup PR when more functionality gets added if you prefer.